internal/tests/TopicBrowsePermissionsResource/topic_browse_permissions_resource_test.go

```diff
@@ -0,0 +1,37 @@
+package TopicBrowsePermissionsResource
+
+import (
+	. "axual.com/terraform-provider-axual/internal/tests"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestTopicBrowsePermissionsResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: GetProviderConfig(t).ProtoV6ProviderFactories,
+		ExternalProviders:        GetProviderConfig(t).ExternalProviders,
+
+		Steps: []resource.TestStep{
+			{
+				Config: GetProvider() + GetFile("axual_topic_browse_permissions_initial.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("axual_topic_browse_permissions.tf-test-topic-browse-permissions", "users.0", "axual_user.ben", "id"),
+					resource.TestCheckResourceAttrPair("axual_topic_browse_permissions.tf-test-topic-browse-permissions", "groups.0", "axual_group.team-group2", "id"),
+				),
+			},
+			{
+				Config: GetProvider() + GetFile("axual_topic_browse_permissions_updated.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("axual_topic_browse_permissions.tf-test-topic-browse-permissions", "users.0", "axual_user.chris", "id"),
+					resource.TestCheckResourceAttrPair("axual_topic_browse_permissions.tf-test-topic-browse-permissions", "groups.0", "axual_group.team-group3", "id"),
+				),
+			},
+			{
```

Comment:
Can we add an additional test scenario for missing required fields(users and groups)?

Reply:
I added this test but in general, e2e tests are not a great place to put error assertions, unit tests are better for that. And in this case Terraform Provider is not really responsible for this error, API is, so API unit tests should be tested for this. We are currently doing API unit test in Terraform e2e test.